pynicotine/gtkgui/search.py

```diff
@@ -436,57 +428,101 @@ def __init__(self, searches, text, token, mode, mode_label, show_page):
+                    "column_type": "number",
+                    "title": _("Speed"),
+                    "width": 120,
+                    "sort_column": "speed_data",
+                    "sensitive_column": "free_slot_data"
+                },
+                "in_queue": {
+                    "column_type": "number",
+                    "title": _("In Queue"),
+                    "width": 110,
+                    "sort_column": "in_queue_data",
+                    "sensitive_column": "free_slot_data"
+                },
+                "folder": {
+                    "column_type": "text",
+                    "title": _("Folder"),
+                    "width": 200,
+                    "expand_column": True,
+                    "sensitive_column": "free_slot_data",
+                    "tooltip_callback": self.on_file_path_tooltip
+                },
+                "file_type": {
+                    "column_type": "icon",
+                    "title": _("File Type"),
+                    "width": 40,
+                    "hide_header": True,
+                    "sensitive_column": "free_slot_data"
+                },
+                "filename": {
+                    "column_type": "text",
+                    "title": _("Filename"),
+                    "width": 200,
+                    "expand_column": True,
+                    "sensitive_column": "free_slot_data",
+                    "tooltip_callback": self.on_file_path_tooltip
+                },
+                "size": {
+                    "column_type": "number",
+                    "title": _("Size"),
+                    "width": 180,
+                    "sort_column": "size_data",
+                    "sensitive_column": "free_slot_data"
+                },
+                "quality": {
+                    "column_type": "number",
+                    "title": _("Quality"),
+                    "width": 150,
+                    "sort_column": "bitrate_data",
+                    "sensitive_column": "free_slot_data"
+                },
+                "length": {
+                    "column_type": "number",
+                    "title": _("Duration"),
+                    "width": 100,
+                    "sort_column": "length_data",
+                    "sensitive_column": "free_slot_data"
+                },
+
+                # Hidden data columns
+                "country_data": {"data_type": str},
+                "speed_data": {"data_type": GObject.TYPE_UINT},
+                "in_queue_data": {"data_type": GObject.TYPE_UINT},
+                "size_data": {"data_type": GObject.TYPE_UINT64},
+                "bitrate_data": {"data_type": GObject.TYPE_UINT},
+                "length_data": {"data_type": GObject.TYPE_UINT},
+                "file_path_data": {"data_type": str},
+                "free_slot_data": {"data_type": bool},
+                "file_attributes_data": {"data_type": GObject.TYPE_PYOBJECT},
+                "id_data": {
```

Comment:
I suppose the optional "ID" column no longer desired, hence why it is moved out of visible columns?

Reply:
Yes, you can now click the column headers three times to revert to this sorting.

Reply:
It works flawless for most of the columns, apart from the Speed column, on the third press it still appears to be in ascending order for some reason.

Reply:
I must have been mistaken. It seems fine now.

Reply:
There might be a subtle bug somewhere, I think I had a similar issue a while ago.

Reply:
> It works flawless for most of the columns, apart from the "Speed" column, on the third press it still appears to be in ascending order for some reason.
> 
> Also the "In Queue" column is affected by the same problem - both ascending and unsorted are the same - so that would be all columns of type "number".

Possibly fixed in https://github.com/nicotine-plus/nicotine-plus/commit/707e187c5434fce0aa4761359cedf9904bdbdbf1